sdk/azidentity/interactive_browser_credential.go

```diff
@@ -73,11 +77,7 @@ func (c *InteractiveBrowserCredential) GetToken(ctx context.Context, opts policy
 		return azcore.AccessToken{Token: ar.AccessToken, ExpiresOn: ar.ExpiresOn.UTC()}, err
 	}
 
-	o := []public.InteractiveAuthOption{}
-	if c.options.RedirectURL != "" {
-		o = append(o, public.WithRedirectURI(c.options.RedirectURL))
-	}
-	ar, err = c.client.AcquireTokenInteractive(ctx, opts.Scopes, o...)
+	ar, err = c.client.AcquireTokenInteractive(ctx, opts.Scopes, public.WithLoginHint(c.options.LoginHint), public.WithRedirectURI(c.options.RedirectURL))
```

Comment:
If `options.LoginHint` and `options.RedirectURL` are empty, these options are just no-op?

Reply:
Yes. The values end up as string fields on a struct, so an explicit empty string is the same as not setting these options. Is that weird? I'm happy to keep the conditionals if they prevent future maintainers from wondering about this.

Reply:
I think it's fine and makes for simpler code on our side, assuming the contract for these APIs doesn't change on the MSAL side (seems unlikely and would show up in our unit testing).